packages/aws-cdk-lib/aws-apigateway/README.md

```diff
@@ -995,6 +995,21 @@ const api = new apigateway.RestApi(this, 'books', {
   cloudWatchRoleRemovalPolicy: cdk.RemovalPolicy.DESTROY,
 });
 ```
+### Deploying to an existing stage
+
+If you want to use an existing stage for a deployment, you can specify the stage name in the deployment as follows:
+```ts
+const restApi = new apigateway.RestApi(this, 'my-api', {
+  deploy: false,
+});
```

Comment:
Seems like existing stage is not created while creating the api in the example. Is below required? If so can we update it in `README.md`

``` 
 deployOptions: {
    stageName: 'dev',
  }
```

Reply:
I believe if `deploy: false` is set then `RestApi` does not automatically create a deployment and stage. As such it's also not possible to set `deployOptions` if `deploy` is set to false. Doing so and running `cdk synth` results in error: `Cannot set 'deployOptions' if 'deploy' is disabled`.

So this example is where the user wants to manually create a deployment and set the stage to one they've already created. I tried deploying it with a API gateway stage called 'dev' in my account and it didn't create a "prod" stage for me but let me know if you found otherwise.

Reply:
Thank you for the explanation. But I'm still not clear on where the existing stage `dev` is defined. I hope  `stageName: 'dev'` is only refering the existing stage which is already available and doesn't create if not available already. Please correct me if I'm wrong. Hence, It could be worth mentioning in the example with the stage (existing) defined that is used in the deployment.

```
const deployment = new apigateway.Deployment(this, 'my-deployment', {
  api: restApi,
  stageName: 'dev',
});
```